### PR TITLE
[pwm] Fix NPE when disabling and improve logging

### PR DESCRIPTION
**Pull request by fwolter**

### CODE REVIEW

File: bundles/org.openhab.automation.pwm/src/main/java/org/openhab/automation/pwm/internal/handler/PWMTriggerHandler.java [hunk 1]
```diff
@@ -242,11 +247,6 @@ public void dispose() {
             localEventSubscriberRegistration.unregister();
         }
 
-        StateMachine localStateMachine = stateMachine;
-        if (localStateMachine != null) {
-            localStateMachine.stop();
```
Comment (lolodomo) [inline bundles/org.openhab.automation.pwm/src/main/java/org/openhab/automation/pwm/internal/handler/PWMTriggerHandler.java:247]: Just to be sure that this is deliberately no more needed

**Review by lolodomo — COMMENTED** [review]
Just one question

File: bundles/org.openhab.automation.pwm/src/main/java/org/openhab/automation/pwm/internal/handler/PWMTriggerHandler.java (hunk 1 above)
Comment (fwolter) [reply bundles/org.openhab.automation.pwm/src/main/java/org/openhab/automation/pwm/internal/handler/PWMTriggerHandler.java:247]: Yes, that caused the exception.

**Review by lolodomo — APPROVED** [review]

**Comment by fwolter** [comment]
Thank you!

**Pull request merged**